backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -336,19 +354,23 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
         initializeCachedBoxes(context)
         declaration.acceptChildrenVoid(this)
 
-        // Note: it is here because it also generates some bitcode.
-        context.objCExport.generate(codegen)
+        withStaticInitializer(declaration.descriptor) {
+            // Note: it is here because it also generates some bitcode.
+            context.objCExport.generate(codegen)
```

Comment:
Btw, did ObjCExport suffered from the same problem?

Reply:
It might have: in `ObjCExportCodeGenerator.createObjectInstanceAdapter` there's a call to `getObjectValue`, but I didn't actually check.

Reply:
Turns out it suffered from the problem too. Added a test for it.